nemo_aligner/algorithms/self_rewarding.py

```diff
@@ -0,0 +1,1559 @@
+                                bad_meta_sample = True
+                                # print("BAD_META_SAMPLE_1")
+
+                            meta_chosen_idx = np.argmax(elo_scores)
+                            meta_reject_idx = np.argmin(elo_scores)
+
+                            chosen_prompt_len = reward_tokens_raw[meta_chosen_idx][1]
+                            chosen_gen_len = reward_tokens_raw[meta_chosen_idx][2]
+                            chosen_tokens = reward_tokens_raw[meta_chosen_idx][0][:chosen_gen_len]
+                            # chosen_score = 0.
+                            reject_prompt_len = reward_tokens_raw[meta_reject_idx][1]
+                            reject_gen_len = reward_tokens_raw[meta_reject_idx][2]
+                            reject_tokens = reward_tokens_raw[meta_reject_idx][0][:reject_gen_len]
+                            # reject_score = 0.
+                            meta_bad_ends = sum(
+                                ~np.array(
+                                    [reward_tokens_raw[meta_chosen_idx][-1], reward_tokens_raw[meta_reject_idx][-1]]
+                                )
+                            )
+
+                            if torch.equal(chosen_tokens, reject_tokens):
+                                bad_meta_sample = True
+                                # print("BAD_META_SAMPLE_2")
+
+                            chosen_score = self.parse_reward_fn(
+                                self.tokenizer.ids_to_text(chosen_tokens[chosen_prompt_len:chosen_gen_len].tolist())
+                            )
+                            reject_score = self.parse_reward_fn(
+                                self.tokenizer.ids_to_text(reject_tokens[reject_prompt_len:reject_gen_len].tolist())
+                            )
+                            # print(f"*** Iteration [ {self.iteration} ] Step [ {self.step} ] META_ACTUAL_REWARDS  CHOSEN[ {chosen_score} ]  REJECT[ {reject_score} ]")
+                            if chosen_score is None or reject_score is None or chosen_score == reject_score:
+                                bad_meta_sample = True
+                                # print("BAD_META_SAMPLE_3")
+
+                            if (
+                                meta_bad_ends == 0
+                                and bad_meta_sample == False
+                                and (
+                                    (cnt_tracker / sum(cnt_tracker).clip(min=1.0))[int(chosen_score)]
+                                    < self.meta_max_relative_pcnt
+                                )
+                                and (
+                                    cnt_tracker[int(chosen_score)]
+                                    < int(
+                                        self.num_steps_per_epoch
+                                        * original_gbs_size
+                                        / ((self.judge_score_high - self.judge_score_low) ** 2)
```

Comment:
Can you please add a comment to explain this condition?

Reply:
Added